content/basics/basics.mdx

```diff
@@ -0,0 +1,327 @@
+    this.animals.push(animal)
+  }
+}
+```
+
+## A React component that observes
+
+We haven't used anything except plain MobX in this example. Now let's
+create a React component to show information for an animal:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const AnimalView = observer(({ animal }: { animal: Animal }) => {
+  return (
+    <div>
+      {animal.name} is of species {animal.species} and is{' '}
+      {animal.isHungry ? 'hungry' : 'not hungry'}.
+    </div>
+  )
+})
+```
+
+> These examples work with `mobx-react-lite` too. `mobx-react` expands
+> `mobx-react-lite` to have support for React class components. If you are
+> creating a new project that only uses React function components, you can use
+> `mobx-react-lite`.
+
+Let's create an instance of `Animal`:
+
+```tsx
+const fred = new Animal('Fred', 'hippo')
+```
+
+We can now render `fred`:
+
+```tsx
+<AnimalView animal={fred} />
+```
+
+The only special thing that `AnimalView` does compared to a normal React
+function component is that it is wrapped in `observer` from `mobx-react`. This
+makes `AnimalView` automatically re-render itself if anything that it observes
+changes. In this case the observable information is the computed property
+`isHungry`, which in turn it based on the observable property `energy`.
+`AnimalView` automatically re-renders itself only if `isHungry` changes. You
+don't need to do anything else to make that happen.
+
+When you wrap your React component in `observer`, it tracks any observable
+property or object you access during rendering it, and reacts to any changes to
+these. It won't react to any changes in observable properties you _don't_
+directly or indirectly use during rendering.
+
+## When do observers observe?
+
+Note that MobX can only observe properties, not simple values like a `string`
+or a `number`. So if you write your component to only receive simple values, it
+does not benefit from observability:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+// Uses observer() but does not benefit from observability
+const NonObservingAnimalView = observer(
+  ({
+    name,
+    species,
+    isHungry,
+  }: {
+    name: string
+    species: string
+    isHungry: boolean
+  }) => {
+    return (
+      <div>
+        {name} is of species {species} and is{' '}
+        {isHungry ? 'hungry' : 'not hungry'}.
+      </div>
+    )
+  },
+)
+```
+
+The component `NonObservingAnimalView` works perfectly if it is used within
+another component that receives an `Animal` instance and is also an observer,
+but it re-render each and every time, because MobX cannot see whether
+`name`, `species` or `isHungry` are changed. Our original `AnimalView` _does_
+observe it, because it can track property accesses.
+
+Wrapping a component in `observer` is cheap, so if you use MobX with react,
+it's recommended to just wrap all your components in `observer`, even if the
```

Comment:
I wouldn't say it's "recommended". It's an escape hatch so you don't need to think about it. The recommended is to use https://mobx.js.org/refguide/api.html#computedrequiresreaction-boolean